storage/csv_test.go

```diff
@@ -169,3 +194,86 @@ func TestCSVPersistComposite(t *testing.T) {
 			"42\n",
 		string(otherWritten))
 }
+
+func TestCSVPersistTime(t *testing.T) {
+	// use time.Now since the default string value includes the monotonic clock, so we can test it is not present in csv output
+	now := time.Now()
+
+	tm := &TimeModel{
+		Height:    42,
+		Processed: now,
+	}
+
+	dir, err := ioutil.TempDir("", t.Name())
+	require.NoError(t, err)
+
+	defer os.RemoveAll(dir)
+
+	st, err := NewCSVStorage(dir)
+	require.NoError(t, err)
+
+	err = st.PersistBatch(context.Background(), tm)
+	require.NoError(t, err)
+
+	written, err := ioutil.ReadFile(filepath.Join(dir, "time_models.csv"))
+	require.NoError(t, err)
+	assert.EqualValues(t,
+		"height,processed\n"+
+			"42,"+now.Format(PostgresTimestampFormat)+"\n",
+		string(written))
+}
+
+func TestCSVPersistInterfaceNil(t *testing.T) {
+	tm := &InterfaceModel{
+		Height: 42,
+		Value:  nil,
+	}
+
+	dir, err := ioutil.TempDir("", t.Name())
+	require.NoError(t, err)
+
+	defer os.RemoveAll(dir)
+
+	st, err := NewCSVStorage(dir)
+	require.NoError(t, err)
+
+	err = st.PersistBatch(context.Background(), tm)
+	require.NoError(t, err)
+
+	written, err := ioutil.ReadFile(filepath.Join(dir, "interface_models.csv"))
+	require.NoError(t, err)
+	assert.EqualValues(t,
+		"height,value\n"+
+			"42,NULL\n",
+		string(written))
+}
+
+func TestCSVPersistInterfaceValue(t *testing.T) {
+	tm := &InterfaceModel{
+		Height: 42,
+		Value: []*ProcessingError{
+			{
+				Source: "some task",
+				Error:  fmt.Errorf("processing error"),
+			},
+		},
+	}
+
+	dir, err := ioutil.TempDir("", t.Name())
+	require.NoError(t, err)
+
+	defer os.RemoveAll(dir)
+
+	st, err := NewCSVStorage(dir)
+	require.NoError(t, err)
+
+	err = st.PersistBatch(context.Background(), tm)
+	require.NoError(t, err)
+
+	written, err := ioutil.ReadFile(filepath.Join(dir, "interface_models.csv"))
+	require.NoError(t, err)
+	assert.EqualValues(t,
+		"height,value\n"+
+			"42,\"[{\"\"Source\"\":\"\"some task\"\",\"\"Error\"\":{}}]\"\n",
```

Comment:
This assertion does not match the Error contained in `tm`.

Reply:
Will look into that